go/kbfs/libkbfs/folder_block_ops.go

```diff
@@ -216,7 +216,9 @@ type folderBlockOps struct {
 
 	// dirtyDirs track which directories are currently dirty in this
 	// TLF.
-	dirtyDirs map[data.BlockPointer][]data.BlockInfo
+	dirtyDirs          map[data.BlockPointer][]data.BlockInfo
+	dirtyDirsSyncing   bool
+	deferredDirUpdates []func(lState *kbfssync.LockState) error
```

Comment:
Why did you decide to make this a slice of arbitrary functions? It doesn't seem like the wrong choice per se, but it seems hard to debug.

Reply:
The place that calls the function doesn't have all the state it needs about nodes and such, so it's either make a closure like this, or buildout an explicit param struct that saves all the arguments and pass that around instead.  Either seems valid to me, and this was easier and in line with other defer-like functions in this file.  Not sure it impacts debuggability but happy to discuss more if you feel strongly about it.